src/text-editor.coffee

```diff
@@ -3088,11 +3083,40 @@ class TextEditor extends Model
     Grim.deprecate("This is now a view method. Call TextEditorElement::getWidth instead.")
     @displayBuffer.getWidth()
 
-  getScrollRow: -> @scrollRow
-  setScrollRow: (@scrollRow) ->
+  # Experimental: Scroll the editor such that the given screen row is at the
+  # top of the visible area.
+  setFirstVisibleScreenRow: (screenRow, fromView) ->
+    unless fromView
+      maxScreenRow = @getLineCount() - 1
+      unless @config.get('editor.scrollPastEnd')
+        height = @displayBuffer.getHeight()
+        lineHeightInPixels = @displayBuffer.getLineHeightInPixels()
+        if height? and lineHeightInPixels?
+          maxScreenRow -= Math.floor(height / lineHeightInPixels)
+      screenRow = Math.max(Math.min(screenRow, maxScreenRow), 0)
 
-  getScrollColumn: -> @scrollColumn
-  setScrollColumn: (@scrollColumn) ->
+    unless screenRow is @firstVisibleScreenRow
+      @emitter.emit 'did-change-first-visible-screen-row', screenRow unless fromView
+      @firstVisibleScreenRow = screenRow
```

Comment:
:+1: Thanks @t9md. This method isn't intended for public consumption, but that would still be better style.